packages/framework/presence/src/test/presenceManager.spec.ts

```diff
@@ -113,114 +104,216 @@ describe("Presence", () => {
+							const attendees = presence.getAttendees();
+							assert(
+								attendees.has(knownAttendee),
+								"getAttendees set does not contain attendee",
+							);
+							assert.equal(
+								knownAttendee.getConnectionStatus(),
+								SessionClientStatus.Disconnected,
+								"getAttendees set contains attendee with wrong status",
+							);
 						});
-					}
+					});
+
+					it('is not announced via `attendeeJoined` when already "Connected"', () => {
+						// Setup
+						afterCleanUp.push(
+							presence.events.on("attendeeJoined", () => {
+								assert.fail("No attendee should be announced in beforeEach");
+							}),
+						);
 
-					it("is available from `getAttendees`", () => {
+						// Act - simulate join message from client
+						presence.processSignal("", initialAttendeeSignal, false);
+					});
+
+					describe("has their connection removed", () => {
 						// Setup
-						assert(newAttendee !== undefined, "No attendee was set in beforeEach");
+						let disconnectedAttendee: ISessionClient | undefined;
+						beforeEach(() => {
+							afterCleanUp.push(
+								presence.events.on("attendeeDisconnected", (attendee) => {
+									assert(
+										disconnectedAttendee === undefined,
+										"Only one attendee should be disconnected",
+									);
+									disconnectedAttendee = attendee;
+								}),
+							);
+
+							// Act - remove client connection id
+							presence.removeClientConnectionId(initialAttendeeConnectionId);
+
+							it("is announced via `attendeeDisconnected`", () => {
+								assert(
+									disconnectedAttendee !== undefined,
+									"No attendee was disconnected in removeClientConnectionId",
+								);
+								assert.equal(
+									disconnectedAttendee.sessionId,
+									newAttendeeSessionId,
+									"Disconnected attendee has wrong session id",
+								);
+								assert.equal(
+									disconnectedAttendee.getConnectionId(),
+									initialAttendeeConnectionId,
+									"Disconnected attendee has wrong client connection id",
+								);
+								assert.equal(
+									disconnectedAttendee.getConnectionStatus(),
+									SessionClientStatus.Disconnected,
+									"Disconnected attendee has wrong status",
+								);
+							});
+							it('is not announced via `attendeeDisconnected` when already "Disconnected"', () => {
+								assert(
+									disconnectedAttendee !== undefined,
+									"No attendee was disconnected in removeClientConnectionId",
+								);
 
-						// Act
-						const attendees = presence.getAttendees();
-						assert(attendees.has(newAttendee), "getAttendees set does not contain attendee");
+								// Act & Verify - remove client connection id again
+								presence.removeClientConnectionId(initialAttendeeConnectionId);
+							});
+						});
 					});
+				});
 
-					it("is NOT announced when rejoined with same connection (duplicate signal)", () => {
```

Comment:
What happened to the duplicate signal test case?

Reply:
Added it back!